### PR TITLE
chore(ci): update CI to run on pull_request event

### PR DESCRIPTION
**Pull request by rwaskiewicz**
## Pull request checklist

Please check if your PR fulfills the following requirements:
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been reviewed and added / updated if needed (for bug fixes / features)
- [ ] Build (`npm run build`) was run locally ~and any changes were pushed~

## Pull request type

 

 

Please check the type of change your PR introduces:
- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [x] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe): Backport of https://github.com/ionic-team/stencil/pull/2934

## What is the current behavior?

Currently, GitHub Actions don't trigger for PRs created from forked repos, only when a push is made directly to the Stencil repo

## What is the new behavior?

update CI to run on pull_request event to run on PRs from forked repos

## Does this introduce a breaking change?

- [ ] Yes
- [x] No

## Other information

- [GitHub Actions Documentation for PRs from Forked Repos](https://docs.github.com/en/actions/reference/events-that-trigger-workflows#pull-request-events-for-forked-repositories)
- [GitHub Actions Documentation for push events](https://docs.github.com/en/actions/reference/events-that-trigger-workflows#push)
> Runs your workflow when someone pushes to a repository branch, which triggers the push event.
(not particularly helpful on its own admittedly)
- I used [this configuration from the Capacitor team](https://github.com/ionic-team/capacitor/blob/main/.github/workflows/ci.yml#L3-L9) as the source of inspiration/start of my documentation reading

### CODE REVIEW

File: .github/workflows/main.yml [hunk 1]
```diff
@@ -1,6 +1,12 @@
 name: Build & Test
 
-on: push
+on:
+  push:
+    branches:
+      - '**'
+  pull_request:
+    branches:
+      - '**'
```
Comment (ltm) [inline .github/workflows/main.yml:9]: Specifying the branches is not necessary in this case:
```suggestion
on:
  push:
  pull_request:
```

Alternatively you can use the shorthand `on: [push, pull_request]`.

Comment (rwaskiewicz) [reply .github/workflows/main.yml]: I went with the shorthand, that 'felt more readable' to me

Comment (rwaskiewicz) [reply .github/workflows/main.yml]: Hmmm but it looks like as-is (I didn't think to check beforehand) these events aren't exclusive. IE right now, there's a pull_request _and_ push set of checks running. Let me see what we can do about that

**Comment by rwaskiewicz** [comment]
@ltm I updated this to be explicit to run CI on our protected branches, and once on CI. I think that achieves what we effectively want - a clean CI run on the `master` (and `v3.0.0-dev`) branches every time we merge to them, but doesn't run both the ubuntu & windows builds 2x each on a PR. At this point, I think I prefer being "explicit" about `branches` in the pull_request event, even if it just is a glob.  Let me know what you think.

**Review by ltm — APPROVED** [review]

**Pull request merged**